### ISSUE TITLE
Firestore emulator does not adhere to specified limitations

### ISSUE THREAD

**Issue by mad-it** [firebase/firebase-tools#1899]
### [REQUIRED] Environment info

**firebase-tools:** 7.11.0

**Platform:** Ubuntu

### [REQUIRED] Test case

According to the documentation [here](https://firebase.google.com/docs/firestore/quotas#collections_documents_and_fields) there are limitations imposed on for example document sizes. These limitations are not adhered to by the firestore emulator.

### [REQUIRED] Steps to reproduce

Create a document in any collection and try fitting an object with a size > 1MB.

### [REQUIRED] Expected behavior

A GRPC error should have been thrown by the emulator.

### [REQUIRED] Actual behavior

The emulator accepts the value even though it exceeds the specified limits.

**Comment by samtstern** [comment]
@mad-it thanks for reporting!  Sounds like a bug to me, will let @ryanpbrewster comment and confirm.

**Comment by ryanpbrewster** [comment]
Good catch, we don't enforce this limit at all in the emulator. Adding support for it, will update this issue when the next release goes out.

**Comment by mad-it** [comment]
@ryanpbrewster  Please consider replicating for all the limitations imposed by Firestore and not just on document size. The community would greatly benefit from this if both the emulator and product are aligned.

**Comment by ryanpbrewster** [comment]
Some of the production limits will be difficult to enforce in the emulator, but we can enforce the majority of them and document any exceptions.

I've filed an internal tracking issue for this work, and for the documentation associated with it.
